packages/launchpad/src/layouts/HeaderBar.spec.tsx

```diff
@@ -0,0 +1,57 @@
+import { HeaderBarFragmentDoc } from '../generated/graphql-test'
+import HeaderBar from './HeaderBar.vue'
+
+describe('<HeaderBar />', () => {
+  it('renders with functional browser menu when show-browsers prop is true', () => {
+    cy.viewport(1000, 750)
+    cy.mountFragment(HeaderBarFragmentDoc, {
+      type: (ctx) => {
+        return {
+          ...ctx.stubQuery,
+        }
+      },
+      render: (gqlVal) => <div class="resize overflow-auto border-current border-1 h-700px"><HeaderBar gql={gqlVal} show-browsers={true} /></div>,
+    })
+
+    cy.contains('button', 'Electron v73')
+    .should('be.visible')
+    .click()
+
+    cy.contains('Edge Canary')
+    .should('be.visible')
+  }),
+
+  it('renders without browser menu by default and other items work', () => {
+    cy.viewport(1000, 750)
+    cy.mountFragment(HeaderBarFragmentDoc, {
+      type: (ctx) => {
+        return {
+          ...ctx.stubQuery,
+        }
+      },
+      render: (gqlVal) => <div class="resize overflow-auto border-current border-1 h-700px"><HeaderBar gql={gqlVal} /></div>,
+    })
+
+    // TODO: These selectors could be a little more stable but let's revisit when everything is wired up.
```

Comment:
Yeah, I particularly like querying for the strings we should show from the `defaultMessages` exported from `@cy/i18n`.

Reply:
This is much better. I grabbed what I can from i18n and used data attrs for the rest.